php/class-post-type.php

```diff
@@ -675,4 +685,140 @@ public function hide_disabled_publishing_actions( $post ) {
 		</style>
 		<?php
 	}
+
+	/**
+	 * Add snapshot bulk actions.
+	 *
+	 * @param array $bulk_actions actions.
+	 *
+	 * @return mixed
+	 */
+	public function add_snapshot_bulk_actions( $bulk_actions ) {
+		$bulk_actions['merge_snapshot'] = __( 'Merge Snapshot', 'customize-snapshots' );
+		return $bulk_actions;
+	}
+
+	/**
+	 * Handle bulk actions.
+	 *
+	 * @param string $redirect_to url to redirect to.
+	 * @param string $do_action   current action.
+	 * @param array  $post_ids    post ids.
+	 *
+	 * @return string url.
+	 */
+	public function handle_snapshot_bulk_actions( $redirect_to, $do_action, $post_ids ) {
+		if ( 'merge_snapshot' !== $do_action ) {
+			return $redirect_to;
+		}
+		$posts = array_map( 'get_post', $post_ids );
+		if ( count( $posts ) <= 1 ) {
+			return empty( $redirect_to ) ? add_query_arg( array( 'merge-error' => 1 ) ) : add_query_arg( array( 'merge-error' => 1 ), $redirect_to );
+		}
+
+		$allowed_status = array( 'draft', 'future' );
+		$posts = array_filter( $posts, function( $p ) use ( $allowed_status ) {
+			return in_array( $p->post_status, $allowed_status, true );
+		} );
+
+		if ( count( $posts ) <= 1 ) {
+			return empty( $redirect_to ) ? add_query_arg( array( 'merge-error' => 2 ) ) : add_query_arg( array( 'merge-error' => 2 ), $redirect_to );
+		}
+
+		usort( $posts, function( $a, $b ) {
+			$compare_a = $a->post_modified;
+			$compare_b = $b->post_modified;
+			if ( '0000-00-00 00:00:00' === $compare_a ) {
+				$compare_a = $a->post_date;
+			}
+			if ( '0000-00-00 00:00:00' === $compare_b ) {
+				$compare_a = $b->post_date;
+			}
+			return strtotime( $compare_a ) - strtotime( $compare_b );
+		} );
+
+		foreach ( $posts as $post ) {
+			$post->post_content = $this->get_post_content( $post );
```

Comment:
I think the data parsed from the `post_content` should be put into another array instead of stuffed back into `$post`. So maybe:

```php
$snapshots_data = array();
foreach ( $posts as $post ) {
    $snapshots_data[] = $this->get_post_content( $post );
}
```

And use `$snapshots_data` instead of `$content`.